src/adapter/src/coord/appends.rs

```diff
@@ -396,6 +409,21 @@ impl Coordinator {
         self.trigger_group_commit();
     }
 
+    /// Submit a write to be executed during the next group commit and triggers a group commit.
+    /// Returns a Future that resolves when the write has completed.
+    pub(crate) fn submit_builtin_table_updates<'a>(
+        &'a mut self,
+        updates: Vec<BuiltinTableUpdate>,
+    ) -> BoxFuture<'static, ()> {
+        let (tx, rx) = oneshot::channel();
+        self.pending_writes.push(PendingWriteTxn::System {
+            updates,
+            source: BuiltinTableUpdateSource::Background(tx),
+        });
+        self.trigger_group_commit();
+        Box::pin(rx.map(|_| ()))
+    }
```

Comment:
A bit of bike-shedding but it's not clear what the difference between `submit_builtin_table_updates` and `send_builtin_table_updates` are by their names. Perhaps we can come up with better names?

Reply:
I renamed `submit_builtin_table_updates` to `send_builtin_table_updates_notify`, I also thought about `send_builtin_table_updates_non_blocking`, any thoughts?

Reply:
Maybe something like `execute_builtin_table_updates_immediately` (maybe omit immediately) and `submit_builtin_table_updates`?

Reply:
Updated the naming to use `_blocking` and `_defer` prefixes!